packages/build-info/src/build-systems/nx.ts

```diff
@@ -73,19 +101,43 @@ export class Nx extends BaseBuildTool {
       return null
     }
 
-    return this.getOutputDirFromProjectJSON(packagePath)
+    return this.getOutputFromTarget(packagePath)
   }
 
-  async getOutputDirFromProjectJSON(packagePath: string): Promise<string | null> {
+  /** Retrieve the overridden port of the nx executor for integrated setups */
+  async getPort(packagePath: string): Promise<number | null> {
+    // only nx integrated has the `project.json`
+    if (!this.isIntegrated) {
+      return null
+    }
+    const targets = this.targets.get(packagePath)?.filter((t) => NPM_DEV_SCRIPTS.includes(t.name)) || []
+    const executor = targets[0] && ('executor' in targets[0] ? targets[0].executor : targets[0].builder)
+
+    switch (executor) {
+      case '@nxtensions/astro:dev':
+        return 3000
+      case '@nrwl/next:server':
+      case '@nx/webpack:dev-server':
+      case '@angular-devkit/build-angular:dev-server':
+        return 4200
+      case '@nx-plus/vue:dev-server':
+        return 8000
+      default:
+        this.project.report({
+          name: 'UndetectedExecutor',
+          message: `Undetected executor for Nx integrated: ${executor}`,
+        })
+        return null
```

Comment:
@juristr I came up now with a simple check for the executor string.

If you know some other executors would love if you could comment it.